### PR TITLE
Module.Wat2Wasm

### PR DESCRIPTION
**Pull request by martindevans**
Added `Wat2Wasm` method, exposing the wasmtime `wat2wasm` function.

### CODE REVIEW

File: src/Module.cs [hunk 1]
```diff
@@ -316,6 +316,43 @@ public static Module DeserializeFile(Engine engine, string name, string path)
             return new Module(handle, name);
         }
 
+        /// <summary>
+        /// Convert WAT (Web Assembly Text) into WASM bytes
+        /// </summary>
+        /// <param name="text"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static byte[] Wat2Wasm(string text)
```
Comment (peterhuene) [inline src/Module.cs:325]: I don't think we have to limit ourselves to the underlying C API name here. Perhaps `ConvertText` would be a better name for C# (open to other names too)?

**Comment by martindevans** [comment]
I've made those two suggested changes :)

File: src/Module.cs [hunk 2]
```diff
@@ -316,6 +316,44 @@ public static Module DeserializeFile(Engine engine, string name, string path)
             return new Module(handle, name);
         }
 
+        /// <summary>
+        /// Convert WAT (Web Assembly Text) into WASM bytes
+        /// </summary>
+        /// <param name="wat">A string containing WAT (Web Assembly Text)</param>
+        /// <returns></returns>
```
Comment (peterhuene) [inline src/Module.cs:323]:
```suggestion
        /// <returns>Returns a byte array containing the WebAssembly module represented by the given text.</returns>
```

**Review by peterhuene — APPROVED** [review]

File: src/Module.cs (hunk 2 above)
Comment (martindevans) [reply src/Module.cs:323]: Oops, I should have filled that one in myself. Sorry about that 🤦‍♂️

**Comment by peterhuene** [comment]
@martindevans thanks for this! I'll merge when CI goes green.

**Pull request merged**